frontend/packages/console-dynamic-plugin-sdk/src/extensions/yaml-templates.ts

```diff
@@ -0,0 +1,24 @@
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+import { K8sKind } from 'public/module/k8s/types';
+
+export namespace ExtensionProperties {
+  export type YAMLTemplate = {
+    /** Model associated with the template. */
+    model: K8sKind;
```

Comment:
To be consistent with other model references in the new extension format:
```suggestion
model: {
  group: string,
  version: string,
  kind: string,
}
```